From efc404381d542750e3a1bcc196cf1a518e11df79 Mon Sep 17 00:00:00 2001
From: Ezra Buehler <ezra.buehler@husqvarnagroup.com>
Date: Thu, 11 Jul 2024 17:49:36 +0200
Subject: [PATCH] bootcount: Allow boot counter to be enabled for AT91SAM9X5

The AT91 boot count driver also works for the AT91SAM9X5 SoC family.

This has been tested on the AT91SAM9G25-based GARDENA smart Gateway.

Signed-off-by: Ezra Buehler <ezra.buehler@husqvarnagroup.com>
Upstream-Status: Pending
---
 drivers/bootcount/Kconfig | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/bootcount/Kconfig b/drivers/bootcount/Kconfig
index 3c56253b1eaa..7281ab35543b 100644
--- a/drivers/bootcount/Kconfig
+++ b/drivers/bootcount/Kconfig
@@ -13,7 +13,7 @@ if BOOTCOUNT_LIMIT
 choice
 	prompt "Boot count device"
 	default BOOTCOUNT_AM33XX if AM33XX || SOC_DA8XX
-	default BOOTCOUNT_AT91 if AT91SAM9XE
+	default BOOTCOUNT_AT91 if AT91SAM9XE || AT91SAM9X5
 	default BOOTCOUNT_GENERIC
 
 config BOOTCOUNT_GENERIC
@@ -80,8 +80,8 @@ config BOOTCOUNT_RAM
 	  due to short power loss or holding a system in RESET.
 
 config BOOTCOUNT_AT91
-	bool "Boot counter for Atmel AT91SAM9XE"
-	depends on AT91SAM9XE
+	bool "Boot counter for Atmel AT91SAM9XX"
+	depends on AT91SAM9XE || AT91SAM9X5
 
 config DM_BOOTCOUNT
         bool "Boot counter in a device-model device"
-- 
2.43.0

